static/app/components/core/compactSelect/listBox/index.tsx

```diff
@@ -163,6 +167,7 @@ export function ListBox({
       <ListWrap
         {...mergeProps(listBoxProps, props)}
         onKeyDown={onKeyDown}
+        onMouseLeave={onMouseLeave}
```

Comment:
we are now unconditionally overwriting `onMouseLeave` passed in by the parent or that potentially gets created by `useListBox`. What we need to do is:

- extract `mergeProps`:
```
const mergedProps = mergeProps(listBoxProps, props)
```
- then spread `mergedProps`
```
{...mergedProps}
```
- then have our own `onMouseLeave` function call `mergedProps.onMouseLeave?.()` before doing anything else.

Reply:
Good catch, I've fixed it the way you've suggested 🙏 Thanks @TkDodo